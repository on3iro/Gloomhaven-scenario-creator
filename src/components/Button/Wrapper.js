/**
 * Styled container for Button
 *
 * @namespace Button.Wrapper
 * @memberOf Button
  */

import styled from 'styled-components';


const Wrapper = styled.div`
  display: inline-block;
`;

export default Wrapper;
