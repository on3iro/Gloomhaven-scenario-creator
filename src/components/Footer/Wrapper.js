/**
 * Styled container for Footer
 *
 * @namespace Footer.Wrapper
 * @memberOf Footer
  */

import styled from 'styled-components';


const Wrapper = styled.div`
  clear: both;
  min-height: 100px;
  width: 100%;
  margin-top: auto;
  flex: none;
  flex-shrink: 0;
  padding: 30px;
  background-color: #e3e3e3;
`;

export default Wrapper;
