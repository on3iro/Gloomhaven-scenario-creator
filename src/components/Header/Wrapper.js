/**
  * Styled container for Header
  *
  * @namespace Header.Wrapper
  * @memberOf Header
  */

import styled from 'styled-components';


const Wrapper = styled.div`
  margin-bottom: 10px;
  height: 40px;
  width: 100%;
  flex: none;
  background-color: #333;
  color: #fff;
`;

export default Wrapper;
