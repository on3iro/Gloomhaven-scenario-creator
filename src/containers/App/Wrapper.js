/**
  * Styled Wrapper for the whole application
  *
  * @namespace App.Wrapper
  * @memberOf App
  */

import styled from 'styled-components';


const Wrapper = styled.div`
  height: 100%;
  min-height: 100%;
  display: flex;
  flex-direction: column;
`;

export default Wrapper;
