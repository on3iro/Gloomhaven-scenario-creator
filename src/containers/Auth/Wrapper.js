/**
  * Renders a styled div for the HomePage
  *
  * @namespace HomePage.Wrapper
  * @memberOf HomePage
  */

import styled from 'styled-components';


const Wrapper = styled.div`
  margin: 0 auto;
  width: 50%;
`;

export default Wrapper;
