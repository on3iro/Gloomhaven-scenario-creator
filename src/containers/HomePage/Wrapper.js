import styled from 'styled-components';


const Wrapper = styled.section`
  width: 50%;
  margin: 0 auto;
`;

export default Wrapper;
