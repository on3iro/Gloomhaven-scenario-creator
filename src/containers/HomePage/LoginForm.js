/**
  * Renders a basic LoginForm
  *
  * @namespace HomePage.LoginForm
  * @memberOf HomePage
  */

import React, { Component } from 'react';

import H1 from 'components/H1';
import Button from 'components/Button';
import Input from 'components/Input';


class LoginForm extends Component {
  /**
    * Class component LoginForm
    *
    * @class LoginForm
    * @namespace LoginForm
    * @memberOf HomePage.LoginForm
    */

  constructor(props) {
    /**
      * Constructor of the LoginForm
      *
      * @constructs
      * @memberOf HomePage.LoginForm.LoginForm
      */

    super(props);

    this.state = {
      email: '',
      password: '',
    };
  }

  handleInputChange = (event) => {
    /**
      * Handles changes on input elements.
      * Sets the value of the respective input.
      *
      * @method handleInputChange
      * @memberOf HomePage.LoginForm.LoginForm
      * @namespace handleInputChange
      * @param {Object} event - Triggered event
      * @returns {undefined}
      */

    const target = event.target;
    const value = target.type === 'checkbox' ? target.checked : target.value;
    const name = target.name;

    this.setState({
      [name]: value,
    });

    return undefined;
  }

  render() {
    return (
      <form onSubmit={() => {}}>
        <H1>Login</H1>
        <Input
          name="email"
          type="text"
          placeholder="Email"
          value={this.state.email}
          onChange={this.handleInputChange}
        />
        <Input
          name="password"
          type="password"
          placeholder="Password"
          value={this.state.password}
          onChange={this.handleInputChange}
        />
        <Button submit>Login</Button>
      </form>
    );
  }
}

export default LoginForm;
