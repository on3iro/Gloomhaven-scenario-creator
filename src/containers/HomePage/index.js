/**
 * The HomePage/Landing page
  */

import React from 'react';

import Wrapper from './Wrapper';


const HomePage = () => {
  return (
    <Wrapper>
      <h1>Home page...</h1>
    </Wrapper>
  );
};

export default HomePage;
