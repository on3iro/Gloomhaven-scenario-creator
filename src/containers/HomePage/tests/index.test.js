import React from 'react';
import { shallow } from 'enzyme';

import HomePage from '../index';
import Wrapper from '../Wrapper';


describe('<HomePage />', () => {
  it('should render Wrapper', () => {
    const renderedWrapper = shallow(<HomePage />);
    expect(renderedWrapper.find(Wrapper).length).toMatchSnapshot();
  });

  it('should render h1', () => {
    const renderedWrapper = shallow(<HomePage />);
    expect(renderedWrapper.find('h1').length).toMatchSnapshot();
  });
});
