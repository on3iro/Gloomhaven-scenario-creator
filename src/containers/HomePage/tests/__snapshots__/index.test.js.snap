exports[`<HomePage /> should render Wrapper 1`] = `1`;

exports[`<HomePage /> should render h1 1`] = `1`;
